File: Editorial/169_Majority_Element/JavaScript/5_Randomization.js
var majorityElement = function (nums) {
    const majorityCount = Math.floor(nums.length / 2);
    while (true) {
        const candidate = nums[Math.floor(Math.random() * nums.length)];
        const count = nums.reduce(
            (prev, curr) => (curr === candidate ? prev + 1 : prev),
            0,
        );
        if (count > majorityCount) {
            return candidate;
        }
    }
};

File: Editorial/0101-0200/0169_Majority_Element/Go/5_Randomization.go
func majorityElement(nums []int) int {
    majorityCount := len(nums) / 2
    rand.Seed(time.Now().UnixNano())
    for {
        candidate := nums[rand.Intn(len(nums))]
        if countOccurences(nums, candidate) > majorityCount {
            return candidate
        }
    }
}

func countOccurences(nums []int, num int) int {
    count := 0
    for _, value := range nums {
        if value == num {
            count++
        }
    }
    return count
}
